src/components/GoodsList.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { useState } from 'react';
+
+type Props = {
+  goods: string[],
+};
+
+const GoodsList: React.FC<Props> = ({ goods }) => {
+  const [goodsList, setGoodsList] = useState([...goods]);
+  const [goodLength, setGoodLength] = useState(1);
+
+  const list = goodsList.filter(good => good.length >= goodLength)
+    .map((good) => {
+      return (
+        <li
+          key={good}
+        >
+          {good}
+        </li>
+      );
+    });
```

Comment:
I think there's no need to put this in a separate variable, but no strong opinion